src/nucleus/tt.ml

```diff
@@ -135,10 +133,12 @@ and instantiate es ?(lvl=0) ({term=e';assumptions;loc;} as e) =
   in
   match e' with
 
-    | Type -> {e with assumptions}
+    | Type -> {e with assumptions} (* XXX why do we need assumptions here? Type is well formed by itself *)
```

Comment:
`e` has type `t` which is proven equal to `Type` by the assumptions.
Same thing for constants, except presumably it's more common for the type of a constant to be equal to something else than for Type.